nio/api.py

```diff
@@ -615,6 +615,22 @@ def space_get_hierarchy(
 
         return ("GET", Api._build_path(path, query_parameters, "/_matrix/client/v1"))
 
+    @staticmethod
+    def direct_room_list(access_token: str, user_id: str) -> Tuple[str, str]:
```

Comment:
Overall this implementation is good and straightforward!
For future extensibility, and for solving similar types of problems, I'm thinking there might be another useful abstraction to be done:

Some other function along the line of `_get_account_data` which does all of this same implementation, with one exception: it takes an extra parameter `event_type: str`.

then `direct_room_list` can very simply be:

```python
def direct_room_list(access_token: str, user_id: str) -> Tuple[str, str]:
    return _get_account_data(access_token, user_id, "m.direct")
```

This gives us breadcrumbs for the the future ease of implementing the same thing for `m.ignored_user_list`, or `m.widgets`, `m.megolm_backup.v1`, etc.

Reply:
I'm also waffling on whether `direct_room_list` or `list_direct_rooms` is a better name, but I'm not too worried about that rn

Reply:
direct_room_list tries to be similar in name to room_send and friends, but given there's no other direct_* in the library, this was the best I could come up with. I tried list_direct_rooms and it just felt like it stuck out like a sore thumb due to it being even less similarly named, but I'm open to both.

Reply:
I settled on list_direct_rooms since it just feels nicer